src/hotspot/share/prims/vectorSupport.cpp

```diff
@@ -362,13 +362,21 @@ int VectorSupport::vop2ideal(jint id, BasicType bt) {
  */
 
 JVM_ENTRY(jint, VectorSupport_GetMaxLaneCount(JNIEnv *env, jclass vsclazz, jobject clazz)) {
-#ifdef COMPILER2
   oop mirror = JNIHandles::resolve_non_null(clazz);
   if (java_lang_Class::is_primitive(mirror)) {
     BasicType bt = java_lang_Class::primitive_type(mirror);
-    return Matcher::max_vector_size(bt);
-  }
+    int min_lane_count = 64 / type2aelembytes(bt);
```

Comment:
I am uncertain of the units here. Is the numerator in bits and the denominator in bytes?

Reply:
Thanks Paul for your review.

Oops, the numerator should be 8 (bytes).

Updated.
Testing of jdk/incubator/vector (MaxVector=default/8/4 or without C2) is still fine.
Thanks.

Reply:
Thanks, was the test `VectorShapeInitTest` passing prior to the fix of the numerator?
Perhaps we should be testing more directly on `VectorShape.S_Max_BIT.vectorBitSize()` and `VectorShape.preferredShape` ? 
Also, perhaps we can run with C2 disabled, with `-XX:TieredStopAtLevel=3` ?

Reply:
> Thanks, was the test `VectorShapeInitTest` passing prior to the fix of the numerator?
Yes. It also passed with min_lane_count = 64 / type2aelembytes(bt).

> Perhaps we should be testing more directly on `VectorShape.S_Max_BIT.vectorBitSize()` and `VectorShape.preferredShape` ?
Ok.
But it that case, I'd like to just re-use jdk/incubator/vector/PreferredSpeciesTest.java.

> Also, perhaps we can run with C2 disabled, with `-XX:TieredStopAtLevel=3` ?
Fine.
Although this bug wouldn't be triggered with -XX:TieredStopAtLevel=x, it can help test  with C1.

Patch had been updated.
Any comments?
Thanks.

Reply:
Reusing PreferredSpeciesTest is a good idea.

I now realize that C2 needs to be compiled out to trigger some other cases. In that case I think we can remove the execution with `-XX:TieredStopAtLevel=x`. The test will get executed with various HotSpot configurations by the test infrastructure, eventually...

Looks good.

Reply:
> In that case I think we can remove the execution with `-XX:TieredStopAtLevel=x`.

Fixed. Thanks.